core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java

```diff
@@ -76,15 +77,20 @@ protected InodeDirectory getThis() {
    * @param child the inode to add
    */
   public void addChild(Inode<?> child) {
-    mChildren.add(child);
+    synchronized (child) {
```

Comment:
Gene, since there is only one concurrentHashMap inside each index, it is thread-safe without synchronized.